### ISSUE TITLE
Data mode attributes: throw errors when placed on wrong types

### ISSUE THREAD

**Issue by HeyJoel** [cofoundry-cms/cofoundry#237]
As an example case we had an issue come in where a developer had used the wrong type on a `CustomEntityMultiTypeCollection`:

```csharp
[CustomEntityMultiTypeCollection("PROGRM", "RSRVSS")]
public ICollection<int> CustomEntities { get; set; }
```
The correct type would have been `ICollection<CustomEntityIdentity>`, but the error that came back from the admin panel was from the JSON parser. We could improve the communication here by throwing a more informative error.

**Comment by HeyJoel** [comment]
Fixed, will be release in v0.4

**Comment by HeyJoel** [comment]
Thanks. The docs will be updated with the releae of 0.4 which will hopefully be this week.
